### PR TITLE
[MRG] ENH: Custom section colors in Cell.plot_morphology

### PR DESCRIPTION
**Pull request by ntolley**
I'd like to make a few additions to the visualization functions that will make animating HNN simulations more straight forward. The main change is allowing a custom color for each cell section in `Cell.plot_morphology` like so:

```py
from hnn_core import jones_2009_model
net = jones_2009_model()
sections = list(net.cell_types['L5_pyramidal'].sections.keys())
section_color = {sect_name: f'C{idx}' for idx, sect_name in enumerate(sections)}
net.cell_types['L5_pyramidal'].plot_morphology(color=section_color)
```
image

The other addition will be a `plot_network_morphology` function (alternate name suggestions are appreciated!) that simply calls `Cell.plot_morphology` and places the cells in their appropriate 3D position.

With these two functions the code to make animations is pretty slim and could be added as an example to the main website. Let me know what you guys think! @rythorpe @jasmainak @chenghuzi

### CODE REVIEW

File: hnn_core/cell.py [hunk 1]
```diff
@@ -655,13 +655,20 @@ def parconnect_from_src(self, gid_presyn, nc_dict, postsyn,
 
         return nc
 
-    def plot_morphology(self, ax=None, cell_types=None, show=True):
```
Comment (ntolley) [inline hnn_core/cell.py:658]: Also I don't think `cell_types` is used anywhere? @jasmainak

Comment (jasmainak) [reply hnn_core/cell.py:658]: I thought this was a method of `Network` object so you can loop over the `cell_types` in a network objec ... I remember a student was working on this but not sure if it ever got merged

Comment (ntolley) [reply hnn_core/cell.py:658]: Yeah I think that ended up getting dropped, the tests are passing so I think it's safe to remove for now

**Comment by ntolley** [comment]
Honestly the code for plotting the sections with different colors is pretty self contained. I think it would be better to bring this to merge to keep the diff small, and then I'll open the PR for plotting the network with the true cell morhologies

File: hnn_core/viz.py [hunk 2]
```diff
@@ -849,7 +865,8 @@ def plot_cell_morphology(cell, ax, show=True):
             xs.append(pt[0] + dx)
             ys.append(pt[1] + dz)
             zs.append(pt[2] + dy)
-        ax.plot(xs, ys, zs, 'b-', linewidth=linewidth)
```
Comment (ntolley) [inline hnn_core/viz.py:852]: The `'b-'` is redundant now since the color `'b'` is explicitly set when `None` is passed to `color`

**Review by jasmainak — APPROVED** [review]
merge button is yours @rythorpe !

don't forget to update `whats_new.rst` @ntolley

**Comment by rythorpe** [comment]
Looks good @ntolley!

**Comment by ntolley** [comment]
@rythorpe that's exactly right! Since the voltages recorded for each section are stored in a dictionary, it's really easy to convert that to cell colors

**Pull request merged**